src/dev/cpp/utils.cpp

```diff
@@ -249,7 +249,7 @@ namespace  dsn
             // optimization: zero-copy
             if (!blob.buffer_ptr())
             {
-                std::shared_ptr<char> buffer(new char[len]);
+                std::shared_ptr<char> buffer(new char[len], [](char* ptr){ delete []ptr; });
```

Comment:
is there a runtime cost for using customized deallcator if the default one already works?